Code/GraphMol/MolStandardize/Charge.cpp

```diff
@@ -265,9 +265,23 @@ std::pair<unsigned int, std::vector<unsigned int>> *Reionizer::weakestIonized(
 
 Uncharger::Uncharger()
     : pos_h(SmartsToMol("[+,+2,+3,+4;!H0!$(*~[-])]")),
-      pos_noh(SmartsToMol("[+,+2,+3,+4;H0;!$(*~[-])]")),
+      pos_noh(
+          SmartsToMol("[+,+2,+3,+4;H0;!$(*~[-]),$(*(~[-])~[-]),$(*(=O)~[-])]")),
       neg(SmartsToMol("[-!$(*~[+,+2,+3,+4])]")),
-      neg_acid(SmartsToMol("[$([O-][C,P,S]=O),$([n-]1nnnc1),$(n1[n-]nnc1)]")){};
+      neg_acid(SmartsToMol(
+          // carboxylate, carbonate, sulfi(a)te,
+          // and their thio-analogues
+          "[$([O,S;-][C,S]=[O,S]),"
```

Comment:
This pattern will match `[S-][S]=S`. Is that intentional or would it be better to have two patterns: `$([O,S;-][C]=[O,S])` and `$([O-][S]=[O])`?

Reply:
It was intentional. I am conscious many of those species probably do not even exist, but I think they would be formally correct. Don't you?

Reply:
I can see arguments either way and don't have a really strong opinion about it, but I do think it would be nice if the comment were correct. :-)
Maybe worth mentioning that you're intentionally including things that probably don't exist?

Reply:
Not sure the comment is incorrect. That pattern would match hyposulfite, sulfite, sulfate and some thio analogues that certainly exist (e.g., thiosulfite, thiosulfate), plus some potential thio-analogues of dubious existence, but which are formally correct. What would you add to the comment? Unless I am being thick, which is entirely likely given my previous misreading of the recursive SMARTS :)

Reply:
Maybe adding “among other less likely structures” at the end of the comment?